src/data/move.ts

```diff
@@ -10992,7 +11014,7 @@ export function initMoves() {
       .attr(MultiHitAttr, MultiHitType._2),
     new AttackMove(Moves.RAGE_FIST, Type.GHOST, MoveCategory.PHYSICAL, 50, 100, 10, -1, 0, 9)
       .partial() // Counter resets every wave instead of on arena reset
```

Comment:
```suggestion
```
Were confusion self-hits also incrementing the counter? Or was the issue only related to writing the test? If the former replace this with the second suggestion, otherwise if it's working properly go with the first.
```suggestion
      .edgeCase() // Counter incorrectly increases on confusion self-hits
```